docs/internals.rst

```diff
@@ -0,0 +1,332 @@
+
+1. Make sure ``node`` is an executable and ``npm install`` has been done for the Dredd directory.
+2. `Get Python 3 <https://www.python.org/downloads/>`__. `ReadTheDocs <https://readthedocs.org/>`__ build the documentation with Python 3.6, so make sure you have this version.
+3. Create a `virtual environment <https://docs.python.org/3/library/venv.html>`__ and activate it:
+
+   .. code-block:: shell
+
+      python3 -m venv ./venv
+      source ./venv/bin/activate
+
+4. Install dependencies for the docs:
+
+   .. code-block:: shell
+
+      (venv)$ pip install -r docs/requirements.txt
+
+   .. note::
+
+      We are not using `pipenv <https://pipenv.readthedocs.io/>`__ as it is not yet properly supported by ReadTheDocs.
+
+Now you can use following commands:
+
+-  ``npm run docs:lint`` - Checks quality of the documentation (broken internal and external links, reStructuredText markup mistakes, etc.)
+-  ``npm run docs:build`` - Builds the documentation
+-  ``npm run docs:serve`` - Runs live preview of the documentation on ``http://127.0.0.1:8000``
+
+
+Installation on ReadTheDocs
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The final documentation gets published by `ReadTheDocs <https://readthedocs.org/>`__. Because the documentation needs some of the npm dependencies installed and ReadTheDocs do not support this in their default build environment, we force their latest build image, which includes Node.js out of the box, in the ``readthedocs.yml``. In the Sphinx' configuration file, ``docs/conf.py``, we make sure ``npm install`` is executed on ReadTheDocs.
+
+
+Writing documentation
+~~~~~~~~~~~~~~~~~~~~~
+
+-  Read the `reStructuredText primer <http://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html>`_
+-  No explicit newlines, please - write each paragraph as a single long line and turn on word wrap in your editor
+-  Explicit is better than implicit:
+
+    - Bad: ``npm i -g``
+    - Good: ``npm install --global``
+
+-  When using Dredd's long CLI options in tests or documentation, please always use the notation with ``=`` wherever possible:
+
+    - Bad: ``--path /dev/null``
+    - Good: ``--path=/dev/null``
+
+   While both should work, the version with ``=`` feels more like standard GNU-style long options and it makes arrays of arguments for ``spawn`` more readable.
+-  Do not `title case <https://en.wikipedia.org/wiki/Letter_case#Headings_and_publication_titles>`__ headings, life's too short to spend it figuring out title casing correctly
+-  Using ``127.0.0.1`` (in code, tests, documentation) is preferred over ``localhost`` (see `#586 <https://github.com/apiaryio/dredd/issues/586>`__)
+-  Be consistent
+
+
+Sphinx extensions
+~~~~~~~~~~~~~~~~~
+
+There are several extensions to Sphinx, which add custom directives to the reStructuredText syntax:
+
+- ``.. cli-options:: ./path/to/file.json`` - allows to automatically generate documentation of Dredd's CLI options from the JSON file which specifies them
```

Comment:
and ghissue https://github.com/apiaryio/dredd/pull/1127 ?

Reply:
Yeah, well, I wrote the new contribution docs three weeks before the ghissue extension has existed 😄

But it's a good point I'll be probably forgetting to keep this list updated 😞

Reply:
Yeah, and it's also in separate PR so wasn't nessecerily saying this PR blocks it (I just assumed the other PR would be merged already when I wrote it). Just wanted to point it out so it wasn't forgot